cogs/Voting.py

```diff
@@ -0,0 +1,421 @@
+#!/usr/bin/env python
+
+"""
+Koala Bot Base Cog code and additional base cog functions
+Commented using reStructuredText (reST)
+"""
+# Futures
+
+# Built-in/Generic Imports
+import asyncio
+import time
+
+from dotenv import load_dotenv
+
+# Libs
+import discord
+from discord.ext import commands
+
+# Own modules
+import KoalaBot
+from utils import KoalaDBManager
+
+# Constants
+load_dotenv()
+emote_reference = {1: "1️⃣", 2: "2️⃣", 3: "3️⃣",
+                   4: "4️⃣", 5: "5️⃣", 6: "6️⃣",
+                   7: "7️⃣", 8: "8️⃣", 9: "9️⃣", 10: "🔟"}
+reverse_reference = {v: k for k, v in emote_reference.items()}
+
+
+# Variables
+
+
+def is_vote_caller():
+    async def predicate(ctx):
+        cog = ctx.command.cog
+        if KoalaBot.is_dm_channel(ctx):
+            return False
+        return ctx.author.id in cog.vote_manager.active_votes.keys() and cog.vote_manager.active_votes[
+            ctx.author.id].target_server == ctx.guild.id
+
+    return commands.check(predicate)
+
+
+def test_is_admin(ctx):
+    return KoalaBot.is_admin(ctx) or ctx.author.id == 135496683009081345
+
+
+class Voting(commands.Cog):
+    def __init__(self, bot, db_manager=None):
+        self.bot = bot
+        self.vote_manager = VoteManager()
+        if not db_manager:
+            self.DBManager = KoalaDBManager.KoalaDBManager(KoalaBot.DATABASE_PATH, KoalaBot.DB_KEY)
+            self.DBManager.insert_extension("Voting", 0, True, True)
+
+    @commands.Cog.listener()
+    async def on_raw_reaction_add(self, payload):
+        await self.update_vote_message(payload)
+
+    @commands.Cog.listener()
+    async def on_raw_reaction_remove(self, payload):
+        await self.update_vote_message(payload)
+
+    @staticmethod
+    async def wait_for_message(bot: discord.Client, ctx: commands.Context, timeout: float = 60.0, check=lambda message: message.author == ctx.author):
+        try:
+            msg = await bot.wait_for('message', timeout=timeout, check=check)
+            return msg
+        except Exception:
+            msg = None
+        return msg
+
+    @commands.group(name="vote")
```

Comment:
grouping seems to make it hard to find the commands when using `k!help` is it worth making them separate commands e.g. voteCreate, voteAddOption etc.

Reply:
if you do k!help vote it lists all subcommands, I'll mention that in the docstrings